libnavigation-core/src/main/java/com/mapbox/navigation/module/NavigationModuleProvider.kt

```diff
@@ -0,0 +1,69 @@
+package com.mapbox.navigation.module
+
+import com.mapbox.android.core.location.LocationEngine
+import com.mapbox.android.core.location.LocationEngineRequest
+import com.mapbox.annotation.INSTANTIATION_METHOD_FORMAT
+import com.mapbox.annotation.MODULE_PROVIDER_CLASS_NAME_PREFIX_NAVIGATION
+import com.mapbox.annotation.MODULE_PROVIDER_PACKAGE_NAVIGATION
+import com.mapbox.annotation.navigation.module.MapboxNavigationModuleType
+import com.mapbox.navigation.base.logger.Logger
+import com.mapbox.navigation.base.route.DirectionsSession
+import com.mapbox.navigation.base.route.Router
+import com.mapbox.navigation.base.trip.TripNotification
+import com.mapbox.navigation.base.trip.TripService
+import com.mapbox.navigation.base.trip.TripSession
+import java.lang.reflect.Method
+
+object NavigationModuleProvider {
```

Comment:
We should align the glue classes discovery and extract it to a common place shared with Nav and Maps as we did with annotations. It can be noted as a tailwork.